registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/main/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessor.java

```diff
@@ -435,6 +439,36 @@ private void postAbisIdentification(InternalRegistrationStatusDto registrationSt
 					registrationStatusDto.getRegistrationId(), BioDedupeConstants.ABIS_RESPONSE_NULL);
 
 		} else {
+			if (registrationType.equalsIgnoreCase(RegistrationType.UPDATE.name()) && matchedRegIds.size() == 1
+					&& biometricAutorejectEnable) {
+				Optional<String> optionalMatchedRegId = matchedRegIds.stream().findFirst();
+				String matchedRegId = optionalMatchedRegId.get();
+				String uin = idRepoService.getUinByRid(matchedRegId, utilities.getGetRegProcessorDemographicIdentity());

```

Comment:
If the RID is not latest (lets assume last RID for that user was demographic update), will idrepo still give the correct UIN ?

Reply:
On this line one more question, what if this RID was partially processed and UIN is not yet associated. In that case. what will the method return ?

Reply:
resolved